backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -477,6 +474,21 @@ public RepositoryEntity setRepository(
     return repositoryEntity;
   }
 
+  private void setRepositoryTypeInQueryBuilder(
```

Comment:
this is fine for now , but based not he type we should actually check if the repository is dataset repository.
In future if we add more types of protected repositories this will break.